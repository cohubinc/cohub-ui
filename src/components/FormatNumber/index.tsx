import React from "react";
import NumberFormat from "react-number-format";

export interface IFormatNumberProps {
  value: number;
  thousandSeparator?: boolean;
}

export default class FormatNumber extends React.Component<IFormatNumberProps> {
  static defaultProps = {
    thousandSeparator: true
  };

  render() {
    const { value, ...rest } = this.props;

    return <NumberFormat value={value} displayType={"text"} {...rest} />;
  }
}
