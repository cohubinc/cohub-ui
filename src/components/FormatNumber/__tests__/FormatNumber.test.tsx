import React from "react";
import { mount } from "enzyme";

import FormatNumber from "../";
it("renders FormatNumber without crashing", async () => {
  const wrapper = mount(<FormatNumber value={0} />);

  expect(wrapper).toMatchSnapshot();
});
