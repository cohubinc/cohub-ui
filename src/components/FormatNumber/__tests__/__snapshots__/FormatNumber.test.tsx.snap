// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders FormatNumber without crashing 1`] = `
<FormatNumber
  thousandSeparator={true}
  value={0}
>
  <NumberFormat
    allowEmptyFormatting={false}
    allowLeadingZeros={false}
    allowNegative={true}
    decimalSeparator="."
    displayType="text"
    fixedDecimalScale={false}
    getInputRef={[Function]}
    isAllowed={[Function]}
    isNumericString={false}
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onMouseUp={[Function]}
    onValueChange={[Function]}
    prefix=""
    suffix=""
    thousandSeparator={true}
    thousandsGroupStyle="thousand"
    type="text"
    value={0}
  >
    <span>
      0
    </span>
  </NumberFormat>
</FormatNumber>
`;
